edc-extensions/bpn-validation/bpn-validation-core/src/main/java/org/eclipse/tractusx/edc/validation/businesspartner/functions/BusinessPartnerGroupFunction.java

```diff
@@ -137,18 +136,10 @@ private List<String> parseRightOperand(Object rightValue, PolicyContext context)
         return null;
     }
 
-    private Boolean evaluateIn(BpnGroupHolder bpnGroupHolder) {
+    private Boolean evaluateIsAllOf(BpnGroupHolder bpnGroupHolder) {
         var assigned = bpnGroupHolder.assignedGroups;
-        // checks whether both lists overlap
-        return new HashSet<>(bpnGroupHolder.allowedGroups).containsAll(assigned);
-    }
-
-    private Boolean evaluateNotEquals(BpnGroupHolder bpnGroupHolder) {
-        return !evaluateEquals(bpnGroupHolder);
-    }
-
-    private Boolean evaluateEquals(BpnGroupHolder bpnGroupHolder) {
-        return bpnGroupHolder.allowedGroups.equals(bpnGroupHolder.assignedGroups);
+        var allowed = bpnGroupHolder.allowedGroups;
+        return (assigned.isEmpty() || !allowed.isEmpty()) && new HashSet<>(assigned).containsAll(allowed);
```

Comment:
```suggestion
        return (assigned.isEmpty() || !allowed.isEmpty()) && assigned.containsAll(allowed);
```
as it is already a Set.

Same for `.distinct()` at line 153: not needed anymore

Reply:
changed in [here](https://github.com/eclipse-tractusx/tractusx-edc/pull/1673/commits/8541be9d7cfe82892161b63ab86477d36e10525a).